<a href="https://colab.research.google.com/github/AFELUMO/AFELUMO/blob/main/Unsupervised_Association_Rules_Checkpoint.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Name: Afelumo Temidayo Tom

Title: UNSUPERVISED ASSOCIATION RULES CHECKPOINT

Date: 12/06/2023

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
# Import required libraries
import pandas as pd
import numpy as np
from mlxtend.preprocessing import TransactionEncoder


df_cust_purchases = pd.read_csv('/content/drive/MyDrive/Market_Basket_Optimisation.csv')
df_cust_purchases.head()

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


shrimp    almonds     avocado    vegetables mix green grapes  \
0         burgers  meatballs        eggs               NaN          NaN   
1         chutney        NaN         NaN               NaN          NaN   
2          turkey    avocado         NaN               NaN          NaN   
3   mineral water       milk  energy bar  whole wheat rice    green tea   
4  low fat yogurt        NaN         NaN               NaN          NaN   

  whole weat flour yams cottage cheese energy drink tomato juice  \
0              NaN  NaN            NaN          NaN          NaN   
1              NaN  NaN            NaN          NaN          NaN   
2              NaN  NaN            NaN          NaN          NaN   
3              NaN  NaN            NaN          NaN          NaN   
4              NaN  NaN            NaN          NaN          NaN   

  low fat yogurt green tea honey salad mineral water salmon antioxydant juice  \
0            NaN       NaN   NaN   NaN           NaN    NaN               NaN   
1            NaN       NaN   NaN   NaN           NaN    NaN               NaN   
2            NaN       NaN   NaN   NaN           NaN    NaN               NaN   
3            NaN       NaN   NaN   NaN           NaN    NaN               NaN   
4            NaN       NaN   NaN   NaN           NaN    NaN               NaN   

  frozen smoothie spinach  olive oil  
0             NaN     NaN        NaN  
1             NaN     NaN        NaN  
2             NaN     NaN        NaN  
3             NaN     NaN        NaN  
4             NaN     NaN        NaN

In [ ]:
# Apply one hot encoding on toy dataset
toy_dataset = [['Skirt', 'Sneakers', 'Scarf', 'Pants', 'Hat'],

        ['Sunglasses', 'Skirt', 'Sneakers', 'Pants', 'Hat'],

        ['Dress', 'Sandals', 'Scarf', 'Pants', 'Heels'],

        ['Dress', 'Necklace', 'Earrings', 'Scarf', 'Hat', 'Heels', 'Hat'],

      ['Earrings', 'Skirt', 'Skirt', 'Scarf', 'Shirt', 'Pants']]

te = TransactionEncoder()
te_array = te.fit_transform(toy_dataset)


te_df = pd.DataFrame(te_array, columns = te.columns_)
te_df

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


Dress  Earrings    Hat  Heels  Necklace  Pants  Sandals  Scarf  Shirt  \
0  False     False   True  False     False   True    False   True  False   
1  False     False   True  False     False   True    False  False  False   
2   True     False  False   True     False   True     True   True  False   
3   True      True   True   True      True  False    False   True  False   
4  False      True  False  False     False   True    False   True   True   

   Skirt  Sneakers  Sunglasses  
0   True      True       False  
1   True      True        True  
2  False     False       False  
3  False     False       False  
4   True     False       False

In [ ]:
# Import and run apriori algorithm
from mlxtend.frequent_patterns import apriori
df_freq = apriori(te_df, min_support=0.6)
df_freq

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


support itemsets
0      0.6      (2)
1      0.8      (5)
2      0.8      (7)
3      0.6      (9)
4      0.6   (5, 7)
5      0.6   (9, 5)

In [ ]:
# frequent items
freq_items = apriori(te_df, min_support=0.6, use_colnames= True)
freq_items

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


support        itemsets
0      0.6           (Hat)
1      0.8         (Pants)
2      0.8         (Scarf)
3      0.6         (Skirt)
4      0.6  (Scarf, Pants)
5      0.6  (Pants, Skirt)

In [ ]:
# Display number of freq elements
freq_items['lenght'] = df_freq['itemsets'].apply(lambda x: len(x))
freq_items

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


support        itemsets  lenght
0      0.6           (Hat)       1
1      0.8         (Pants)       1
2      0.8         (Scarf)       1
3      0.6         (Skirt)       1
4      0.6  (Scarf, Pants)       2
5      0.6  (Pants, Skirt)       2

In [ ]:
# Import and run the association rules
from mlxtend.frequent_patterns import association_rules

rules_confidence = association_rules(freq_items, metric='confidence', min_threshold=0.7)
rules_confidence

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


antecedents consequents  antecedent support  consequent support  support  \
0     (Scarf)     (Pants)                 0.8                 0.8      0.6   
1     (Pants)     (Scarf)                 0.8                 0.8      0.6   
2     (Pants)     (Skirt)                 0.8                 0.6      0.6   
3     (Skirt)     (Pants)                 0.6                 0.8      0.6   

   confidence    lift  leverage  conviction  zhangs_metric  
0        0.75  0.9375     -0.04         0.8          -0.25  
1        0.75  0.9375     -0.04         0.8          -0.25  
2        0.75  1.2500      0.12         1.6           1.00  
3        1.00  1.2500      0.12         inf           0.50

In [ ]:
# Association rules with lift metric
rules_lift = association_rules(freq_items, metric= 'lift', min_threshold=0.75)
rules_lift

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


antecedents consequents  antecedent support  consequent support  support  \
0     (Scarf)     (Pants)                 0.8                 0.8      0.6   
1     (Pants)     (Scarf)                 0.8                 0.8      0.6   
2     (Pants)     (Skirt)                 0.8                 0.6      0.6   
3     (Skirt)     (Pants)                 0.6                 0.8      0.6   

   confidence    lift  leverage  conviction  zhangs_metric  
0        0.75  0.9375     -0.04         0.8          -0.25  
1        0.75  0.9375     -0.04         0.8          -0.25  
2        0.75  1.2500      0.12         1.6           1.00  
3        1.00  1.2500      0.12         inf           0.50

From the result obtained from the toy dataset, the frequently bought items are Skirts and Pants. There's a high probability of buying  Pants after buying Skirt as indicated by the Confidence metric(1). Also, lift value of 1.25 means that customers are more likely to buy skirts and pants together rather than buying only one of them.

## Explore Customer Purchases History Dataset

In [ ]:
df_cust_purchases.head()

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


shrimp    almonds     avocado    vegetables mix green grapes  \
0         burgers  meatballs        eggs               NaN          NaN   
1         chutney        NaN         NaN               NaN          NaN   
2          turkey    avocado         NaN               NaN          NaN   
3   mineral water       milk  energy bar  whole wheat rice    green tea   
4  low fat yogurt        NaN         NaN               NaN          NaN   

  whole weat flour yams cottage cheese energy drink tomato juice  \
0              NaN  NaN            NaN          NaN          NaN   
1              NaN  NaN            NaN          NaN          NaN   
2              NaN  NaN            NaN          NaN          NaN   
3              NaN  NaN            NaN          NaN          NaN   
4              NaN  NaN            NaN          NaN          NaN   

  low fat yogurt green tea honey salad mineral water salmon antioxydant juice  \
0            NaN       NaN   NaN   NaN           NaN    NaN               NaN   
1            NaN       NaN   NaN   NaN           NaN    NaN               NaN   
2            NaN       NaN   NaN   NaN           NaN    NaN               NaN   
3            NaN       NaN   NaN   NaN           NaN    NaN               NaN   
4            NaN       NaN   NaN   NaN           NaN    NaN               NaN   

  frozen smoothie spinach  olive oil  
0             NaN     NaN        NaN  
1             NaN     NaN        NaN  
2             NaN     NaN        NaN  
3             NaN     NaN        NaN  
4             NaN     NaN        NaN

In [ ]:
# Display dataset information
df_cust_purchases.info()

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7500 entries, 0 to 7499
Data columns (total 20 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   shrimp             7500 non-null   object 
 1   almonds            5746 non-null   object 
 2   avocado            4388 non-null   object 
 3   vegetables mix     3344 non-null   object 
 4   green grapes       2528 non-null   object 
 5   whole weat flour   1863 non-null   object 
 6   yams               1368 non-null   object 
 7   cottage cheese     980 non-null    object 
 8   energy drink       653 non-null    object 
 9   tomato juice       394 non-null    object 
 10  low fat yogurt     255 non-null    object 
 11  green tea          153 non-null    object 
 12  honey              86 non-null     object 
 13  salad              46 non-null     object 
 14  mineral water      24 non-null     object 
 15  salmon             7 non-null      object 
 16  antioxydant juice  3 non

In [ ]:
# Query dataset to find column contents
df_cust_purchases.query(" shrimp == ['shrimp','almonds','avocado','vegetables mix','green grapes','whole weat flour','yams'] ")

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


shrimp    almonds         avocado vegetables mix green grapes  \
15     shrimp  chocolate         chicken          honey          oil   
47    avocado    muffins             NaN            NaN          NaN   
68     shrimp      pasta             NaN            NaN          NaN   
106    shrimp      pasta   mineral water           soup      avocado   
107    shrimp      pasta            soup           cake  cooking oil   
...       ...        ...             ...            ...          ...   
7397   shrimp      pasta            soup        almonds    chocolate   
7411   shrimp      pasta     ground beef      meatballs          oil   
7441   shrimp      pasta  low fat yogurt            NaN          NaN   
7452  almonds        NaN             NaN            NaN          NaN   
7478   shrimp      pasta   mineral water      olive oil     pancakes   

     whole weat flour            yams  cottage cheese energy drink  \
15        cooking oil  low fat yogurt             NaN          NaN   
47                NaN             NaN             NaN          NaN   
68                NaN             NaN             NaN          NaN   
106              milk       olive oil    green grapes         eggs   
107           chicken      light mayo  low fat yogurt          NaN   
...               ...             ...             ...          ...   
7397      fresh bread             NaN             NaN          NaN   
7411              NaN             NaN             NaN          NaN   
7441              NaN             NaN             NaN          NaN   
7452              NaN             NaN             NaN          NaN   
7478        bug spray  low fat yogurt             NaN          NaN   

     tomato juice low fat yogurt green tea honey salad mineral water salmon  \
15            NaN            NaN       NaN   NaN   NaN           NaN    NaN   
47            NaN            NaN       NaN   NaN   NaN           NaN    NaN   
68            NaN            NaN       NaN   NaN   NaN           NaN    NaN   
106     bug spray            NaN       NaN   NaN   NaN           NaN    NaN   
107           NaN            NaN       NaN   NaN   NaN           NaN    NaN   
...           ...            ...       ...   ...   ...           ...    ...   
7397          NaN            NaN       NaN   NaN   NaN           NaN    NaN   
7411          NaN            NaN       NaN   NaN   NaN           NaN    NaN   
7441          NaN            NaN       NaN   NaN   NaN           NaN    NaN   
7452          NaN            NaN       NaN   NaN   NaN           NaN    NaN   
7478          NaN            NaN       NaN   NaN   NaN           NaN    NaN   

     antioxydant juice frozen smoothie spinach  olive oil  
15                 NaN             NaN     NaN        NaN  
47                 NaN             NaN     NaN        NaN  
68                 NaN             NaN     NaN        NaN  
106                NaN             NaN     NaN        NaN  
107                NaN             NaN     NaN        NaN  
...                ...             ...     ...        ...  
7397               NaN             NaN     NaN        NaN  
7411               NaN             NaN     NaN        NaN  
7441               NaN             NaN     NaN        NaN  
7452               NaN             NaN     NaN        NaN  
7478               NaN             NaN     NaN        NaN  

[470 rows x 20 columns]

In [ ]:
# Reshaping dataset to have all items in a single column
x_transactions = []
for index, row in df_cust_purchases.iterrows():
  transaction = [str(item) for item in row if pd.notna(item)]
  x_transactions.append(transaction)

# Convert to dataframe
transaction_df = pd.DataFrame({'Items': x_transactions})
transaction_df.head()

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


Items
0                         [burgers, meatballs, eggs]
1                                          [chutney]
2                                  [turkey, avocado]
3  [mineral water, milk, energy bar, whole wheat ...
4                                   [low fat yogurt]

In [ ]:
# Encode items data using transaction encoder
from mlxtend.preprocessing import TransactionEncoder
te = TransactionEncoder()
encoded_array = te.fit(transaction_df['Items']).transform(transaction_df['Items'])
encoded_df = pd.DataFrame(encoded_array, columns=te.columns_)
encoded_df

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


asparagus  almonds  antioxydant juice  asparagus  avocado  babies food  \
0          False    False              False      False    False        False   
1          False    False              False      False    False        False   
2          False    False              False      False     True        False   
3          False    False              False      False    False        False   
4          False    False              False      False    False        False   
...          ...      ...                ...        ...      ...          ...   
7495       False    False              False      False    False        False   
7496       False    False              False      False    False        False   
7497       False    False              False      False    False        False   
7498       False    False              False      False    False        False   
7499       False    False              False      False    False        False   

      bacon  barbecue sauce  black tea  blueberries  ...  turkey  \
0     False           False      False        False  ...   False   
1     False           False      False        False  ...   False   
2     False           False      False        False  ...    True   
3     False           False      False        False  ...   False   
4     False           False      False        False  ...   False   
...     ...             ...        ...          ...  ...     ...   
7495  False           False      False        False  ...   False   
7496  False           False      False        False  ...   False   
7497  False           False      False        False  ...   False   
7498  False           False      False        False  ...   False   
7499  False           False      False        False  ...   False   

      vegetables mix  water spray  white wine  whole weat flour  \
0              False        False       False             False   
1              False        False       False             False   
2              False        False       False             False   
3              False        False       False             False   
4              False        False       False             False   
...              ...          ...         ...               ...   
7495           False        False       False             False   
7496           False        False       False             False   
7497           False        False       False             False   
7498           False        False       False             False   
7499           False        False       False             False   

      whole wheat pasta  whole wheat rice   yams  yogurt cake  zucchini  
0                 False             False  False        False     False  
1                 False             False  False        False     False  
2                 False             False  False        False     False  
3                 False              True  False        False     False  
4                 False             False  False        False     False  
...                 ...               ...    ...          ...       ...  
7495              False             False  False        False     False  
7496              False             False  False        False     False  
7497              False             False  False        False     False  
7498              False             False  False        False     False  
7499              False             False  False         True     False  

[7500 rows x 120 columns]

In [ ]:
# Applying apriori algorithm
from mlxtend.frequent_patterns import association_rules

freq_items = apriori(encoded_df, min_support=0.01, use_colnames=True)
freq_items.sort_values(by='support', ascending=False)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


support                                  itemsets
46   0.238267                           (mineral water)
19   0.179733                                    (eggs)
63   0.174133                               (spaghetti)
24   0.170933                            (french fries)
13   0.163867                               (chocolate)
..        ...                                       ...
251  0.010133  (spaghetti, french fries, mineral water)
177  0.010133       (frozen vegetables, low fat yogurt)
123  0.010133                         (soup, chocolate)
164  0.010000                    (french fries, shrimp)
170  0.010000      (frozen vegetables, frozen smoothie)

[259 rows x 2 columns]

In [ ]:
# Generate association rules with confidence metric
rules_confidence = association_rules(freq_items, metric = 'confidence', min_threshold=0.4)
rules_confidence.sort_values(by='confidence', ascending=False)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning:

`should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.



antecedents      consequents  antecedent support  \
7              (eggs, ground beef)  (mineral water)            0.020000   
11             (milk, ground beef)  (mineral water)            0.022000   
4         (chocolate, ground beef)  (mineral water)            0.023067   
9        (milk, frozen vegetables)  (mineral water)            0.023600   
2                           (soup)  (mineral water)            0.050533   
16           (pancakes, spaghetti)  (mineral water)            0.025200   
15          (olive oil, spaghetti)  (mineral water)            0.022933   
14               (milk, spaghetti)  (mineral water)            0.035467   
5                (chocolate, milk)  (mineral water)            0.032133   
12        (spaghetti, ground beef)  (mineral water)            0.039200   
10  (frozen vegetables, spaghetti)  (mineral water)            0.027867   
8                     (milk, eggs)  (mineral water)            0.030800   
1                      (olive oil)  (mineral water)            0.065733   
13    (mineral water, ground beef)      (spaghetti)            0.040933   
0                    (ground beef)  (mineral water)            0.098267   
3                (chocolate, eggs)  (mineral water)            0.033200   
6           (chocolate, spaghetti)  (mineral water)            0.039200   

    consequent support   support  confidence      lift  leverage  conviction  \
7             0.238267  0.010133    0.506667  2.126469  0.005368    1.544054   
11            0.238267  0.011067    0.503030  2.111207  0.005825    1.532756   
4             0.238267  0.010933    0.473988  1.989319  0.005437    1.448130   
9             0.238267  0.011067    0.468927  1.968075  0.005444    1.434328   
2             0.238267  0.023067    0.456464  1.915771  0.011026    1.401441   
16            0.238267  0.011467    0.455026  1.909736  0.005462    1.397744   
15            0.238267  0.010267    0.447674  1.878880  0.004802    1.379138   
14            0.238267  0.015733    0.443609  1.861817  0.007283    1.369061   
5             0.238267  0.014000    0.435685  1.828559  0.006344    1.349836   
12            0.238267  0.017067    0.435374  1.827256  0.007727    1.349094   
10            0.238267  0.012000    0.430622  1.807311  0.005360    1.337834   
8             0.238267  0.013067    0.424242  1.780536  0.005728    1.323011   
1             0.238267  0.027467    0.417850  1.753707  0.011805    1.308483   
13            0.174133  0.017067    0.416938  2.394361  0.009939    1.416431   
0             0.238267  0.040933    0.416554  1.748266  0.017520    1.305576   
3             0.238267  0.013467    0.405622  1.702389  0.005556    1.281565   
6             0.238267  0.015867    0.404762  1.698777  0.006527    1.279712   

    zhangs_metric  
7        0.540548  
11       0.538177  
4        0.509058  
9        0.503778  
2        0.503458  
16       0.488682  
15       0.478747  
14       0.479911  
5        0.468165  
12       0.471202  
10       0.459496  
8        0.452302  
1        0.460018  
13       0.607207  
0        0.474647  
3        0.426758  
6        0.428124

In [ ]:
# Generate Association rules with Lift metric
rules_lift = association_rules(freq_items, metric='lift', min_threshold=1.0)
rules_lift.sort_values(by='lift', ascending=False).head(7)

/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py:283: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.
  and should_run_async(code)


antecedents                 consequents  \
217             (herb & pepper)               (ground beef)   
216               (ground beef)             (herb & pepper)   
386  (mineral water, spaghetti)               (ground beef)   
387               (ground beef)  (mineral water, spaghetti)   
399                 (olive oil)  (mineral water, spaghetti)   
398  (mineral water, spaghetti)                 (olive oil)   
194                  (tomatoes)         (frozen vegetables)   

     antecedent support  consequent support   support  confidence      lift  \
217            0.049467            0.098267  0.016000    0.323450  3.291555   
216            0.098267            0.049467  0.016000    0.162822  3.291555   
386            0.059733            0.098267  0.017067    0.285714  2.907540   
387            0.098267            0.059733  0.017067    0.173677  2.907540   
399            0.065733            0.059733  0.010267    0.156187  2.614731   
398            0.059733            0.065733  0.010267    0.171875  2.614731   
194            0.068400            0.095333  0.016133    0.235867  2.474134   

     leverage  conviction  zhangs_metric  
217  0.011139    1.332841       0.732423  
216  0.011139    1.135402       0.772060  
386  0.011197    1.262427       0.697745  
387  0.011197    1.137893       0.727562  
399  0.006340    1.114306       0.661001  
398  0.006340    1.128171       0.656783  
194  0.009613    1.183913       0.639564

## Result Interpretation and Business Plan Suggestion

###Frequent Items:
The frequently bought items from the Supermarket in descending order using a minimum support threshold of 0.08 are: Mineral water, Eggs, Spaghetti, French fries, Chocolate, Green tea, Milk, Ground beef, Frozen vegetables Pancakes, Burgers, Cakes, Cookies.

###Confidence Metric:
Customers who buy Eggs and Ground Beef or milk and Ground beef are likely to buy Mineral water after, with a probability of 0.5. The result also shows that customers who buy combination of other food items always buy Mineral water after.

###Lift Metric:
Customers mostly buy (Ground beef and herb & Pepper) together rather than buying them separately, same applies to (Mineral water & Spaghetti and Ground beef), (Olive oil and Mineral water & Spaghetti), (Tomatoes and Frozen vegetables)

###Business Plan Suggestion to Supermaket:
Always stock up the store with the frequently bought items.
Place Mineral water in a way that it can be easily found from where Ground beef and other food items are located and label appropriately.
Arrange the items customers prefer buying together in one place- (Ground beef and herb & Pepper), (Mineral water & Spaghetti and Ground beef) and possible offer a discounted rate when they buy them together.
